api/type.rb

```diff
@@ -192,6 +192,19 @@ def initialize(name)
       NAME = Api::Type::String.new('name')
     end
 
+    # Represents a fingerprint.  A fingerprint is an output-only
+    # field used for optimistic locking during updates.
+    class Fingerprint < String
+      def validate
+        super
+        @output = true if @output.nil?
```

Comment:
I think you could do `@output ||= true` here.

Reply:
Sorry, that'll behave wrong if output is set to false.

```ruby
irb(main):008:0> a = false
=> false
irb(main):009:0> a ||= true
=> true
irb(main):010:0> a
=> true
```